ngraph/core/src/op/adaptive_avg_pool.cpp

```diff
@@ -6,11 +6,44 @@
 #include "itt.hpp"
 #include "ngraph/attribute_visitor.hpp"
 #include "ngraph/graph_util.hpp"
+#include "ngraph/runtime/reference/adaptive_avg_pool.hpp"
 #include "ngraph/validation_util.hpp"
 
 using namespace std;
 using namespace ngraph;
 
+namespace adaptive_avg_pool
+{
+    template <element::Type_t ET>
+    bool evaluate(const HostTensorPtr& arg0, const HostTensorPtr& out)
+    {
+        runtime::reference::adaptive_avg_pool(
+            arg0->get_data_ptr<ET>(), out->get_data_ptr<ET>(), arg0->get_shape(), out->get_shape());
+        return true;
+    }
+
+    bool evaluate_adaptive_avg_pool(const HostTensorPtr& arg0, const HostTensorPtr& out)
+    {
+        bool rc = true;
+        switch (arg0->get_element_type())
+        {
+            NGRAPH_TYPE_CASE(evaluate_add, i8, arg0, out);
```

Comment:
Do we really need to support all data types?
Each instantiation affects the binary size.

Reply:
Removed redundant data types